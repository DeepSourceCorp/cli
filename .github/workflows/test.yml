name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
