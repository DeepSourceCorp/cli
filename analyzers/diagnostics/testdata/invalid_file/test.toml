name = "2do-Checker"
