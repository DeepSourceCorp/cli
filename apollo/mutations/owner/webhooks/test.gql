mutation testWebhook($webhookId: ID!) {
  testWebhook(input: { webhookId: $webhookId }) {
    ok
  }
}
