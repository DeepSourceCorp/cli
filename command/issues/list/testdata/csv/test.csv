analyzer,issue_code,issue_title,occurence_title,issue_category,path,begin_line,begin_column,end_line,end_column
go,RVV-B0013,Unused method receiver detected,Unused method receiver detected,,deepsource/transformers/queries/get_transformers.go,34,0,34,0
go,RVV-B0013,Unused method receiver detected,Unused method receiver detected,,deepsource/transformers/queries/get_transformers.go,44,0,44,0
