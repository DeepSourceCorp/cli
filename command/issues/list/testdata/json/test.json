{
	"occurences": [
		{
			"analyzer": "go",
			"issue_code": "RVV-B0013",
			"issue_title": "Unused method receiver detected",
			"occurence_title": "Unused method receiver detected",
			"issue_category": "",
			"location": {
				"path": "deepsource/transformers/queries/get_transformers.go",
				"position": {
					"begin": {
						"line": 34,
						"column": 0
					},
					"end": {
						"line": 34,
						"column": 0
					}
				}
			}
		},
		{
			"analyzer": "go",
			"issue_code": "RVV-B0013",
			"issue_title": "Unused method receiver detected",
			"occurence_title": "Unused method receiver detected",
			"issue_category": "",
			"location": {
				"path": "deepsource/transformers/queries/get_transformers.go",
				"position": {
					"begin": {
						"line": 44,
						"column": 0
					},
					"end": {
						"line": 44,
						"column": 0
					}
				}
			}
		}
	],
	"summary": {
		"total_occurences": 2,
		"unique_issues": 1
	}
}