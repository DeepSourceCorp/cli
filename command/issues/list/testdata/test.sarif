{
  "version": "2.1.0",
  "$schema": "https://json.schemastore.org/sarif-2.1.0-rtm.5.json",
  "runs": [
    {
      "tool": {
        "driver": {
          "informationUri": "https://deepsource.io",
          "name": "deepsource",
          "rules": [
            {
              "id": "RVV-B0013",
              "name": "Unused method receiver detected",
              "shortDescription": {
                "text": ""
              },
              "helpUri": "",
              "properties": {
                "category": "",
                "recommended": ""
              }
            }
          ]
        }
      },
      "results": [
        {
          "ruleId": "RVV-B0013",
          "ruleIndex": 0,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "Unused method receiver detected"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "deepsource/transformers/queries/get_transformers.go"
                },
                "region": {
                  "startLine": 34,
                  "endLine": 34
                }
              }
            }
          ]
        },
        {
          "ruleId": "RVV-B0013",
          "ruleIndex": 1,
          "kind": "fail",
          "level": "error",
          "message": {
            "text": "Unused method receiver detected"
          },
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "deepsource/transformers/queries/get_transformers.go"
                },
                "region": {
                  "startLine": 44,
                  "endLine": 44
                }
              }
            }
          ]
        }
      ]
    }
  ]
}
