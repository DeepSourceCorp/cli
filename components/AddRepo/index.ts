import AddRepoModal from './AddRepoModal.vue'
import RepoCard from './RepoCard.vue'
import SyncRepoAlert from './SyncRepoAlert.vue'

export { AddRepoModal, RepoCard, SyncRepoAlert }
