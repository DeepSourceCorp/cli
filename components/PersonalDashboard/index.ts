import Viewer from './Viewer.vue'
import RecommendedIssues from './RecommendedIssues.vue'
import StarredRepoList from './StarredRepoList.vue'
import ActivityFeedList from './ActivityFeedList.vue'

export { Viewer, RecommendedIssues, StarredRepoList, ActivityFeedList }
