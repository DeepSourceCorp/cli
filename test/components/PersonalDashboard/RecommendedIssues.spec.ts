import { render } from '@testing-library/vue'

import { RecommendedIssues } from '~/components/PersonalDashboard'
import { storeModulesGenerator } from '~/test/mocks'

describe('[[ RecommendedIssues ]]', () => {
  const stubs = {
    AutofixFileChooser: true,
    IssueListItem: true,
    upgradeAccountModal: true
  }

  const storeMock = {
    modules: storeModulesGenerator()
  }

  test('renders `RecommendedIssues` with all prop combinations', () => {
    const { html } = render(RecommendedIssues, {
      store: storeMock,
      stubs
    })

    expect(html()).toMatchSnapshot()
  })
})
