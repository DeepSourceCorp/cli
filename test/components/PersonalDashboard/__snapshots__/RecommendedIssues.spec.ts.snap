// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ RecommendedIssues ]] renders \`RecommendedIssues\` with all prop combinations 1`] = `
"<section class=\\"flex flex-col justify-between\\">
  <div class=\\"grid grid-cols-1 gap-2 md:grid-cols-2 md:gap-0 lg:grid-cols-3 xl:grid-cols-4 2xl:grid-cols-5 pb-3 pt-0\\">
    <div class=\\"col-span-full text-vanilla-100\\">
      <div class=\\"flex h-full items-center space-x-2\\"><span class=\\"text-base font-semibold tracking-snug\\">Recommended issues</span>
        <!---->
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"flex-grow p-0\\">
    <div class=\\"space-y-3\\">
      <autofixfilechooser-stub repo-params=\\"[object Object]\\"></autofixfilechooser-stub>
      <!---->
    </div>
  </div>
  <div></div>
</section>"
`;
